OnBaseDocsApi/Controllers/OnBaseDocsController.cs

```diff
@@ -57,7 +61,200 @@ public IHttpActionResult GetFile(int id)
+                    Comment = docAttr.Comment,
+                    IndexKey = docAttr.IndexKey,
+                    Keywords = docAttr.Keywords,
+                    Ext = docExt,
+                    Stream = docStream,
+                    ToStaging = toStaging,
+                };
+                var error = CreateDocument(app, createAttr, out var doc);
+                if (error != null)
+                    return error;
+
+                if (toStaging)
+                {
+                    // We must get the document ID from this thread and not the task thread since
+                    // the document could go out of scope before the task starts.
+                    var docId = doc.ID;
+                    Task.Run(() =>
+                    {
+                        MoveDocumentToWorkflow(docId, finalDocType);
+                    });
+                }
+
+                return DocumentResult(doc, true);
+            });
+        }
+
+        /// <summary>
+        /// CreateDocument returns an error result when an error occurs; null otherwise. If CreateDocument
+        /// return null, then the out parameter doc is guaranteed to be valid.
+        /// </summary>
+        /// <returns>An error result or null. A null result indicates that the document was created successfully.</returns>
+        IHttpActionResult CreateDocument(Application app, DocumentCreateAttributes attr, out Document doc)
+        {
+            doc = null;
+
+            var docType = app.Core.DocumentTypes.Find(attr.DocumentType);
+            if (docType == null)
+                return BadRequestResult($"The DocumentType '{attr.DocumentType}' could not be found.");
+
+            var fileType = app.Core.FileTypes.Find(attr.Ext);
+            if (fileType == null)
+                return BadRequestResult($"The FileType '{attr.Ext}' could not be found.");
+
+            var props = app.Core.Storage.CreateStoreNewDocumentProperties(docType, fileType);
+            if (props == null)
+                return InternalErrorResult($"Unable to create document properties for '{attr.DocumentType}' and '{attr.Ext}'.");
+
+            props.AddKeyword(Global.Config.DocIndexKeyName, attr.IndexKey);
+            props.DocumentDate = DateTime.Now;
+            props.Comment = attr.Comment;
+            props.ExpandKeysets = false;
+
+            if (attr.ToStaging)
+            {
+                // When uploading a document into staging do not add it to the workflow.
+                props.Options = StoreDocumentOptions.SkipWorkflow;
+            }
+
+            if (attr.Keywords != null)
+            {
+                foreach (var kw in attr.Keywords)
+                {
+                    props.AddKeyword(kw.Name, kw.Value);
+                }
+            }
+
+            var pageData = app.Core.Storage.CreatePageData(attr.Stream, attr.Ext);
+            if (pageData == null)
+                InternalErrorResult($"Unable to create page data for '{attr.Ext}'.");
```

Comment:
I guess this should be bad request as well?

Reply:
Yup.